### PR TITLE
feat: add import endpoint

### PR DESCRIPTION
**Pull request by peterdeme**
- Add import endpoint

### CODE REVIEW

**Comment by peterdeme** [comment]
@gumuz  👀

File: import.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	ID        string               `json:"id"`
+	State     string               `json:"state"`
+	UpdatedAt time.Time            `json:"updated_at"`
+	Result    interface{}          `json:"result"`
+	Size      *int                 `json:"size"`
+}
+
+type CreateImportResponse struct {
+	ImportTask *ImportTask `json:"import_task"`
+	UploadUrl  string      `json:"upload_url"`
```
[the remark this answers is not in the document]
Comment (peterdeme) [reply import.go:28]: 🥳 cool, it works

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,19 +16,17 @@ jobs:
...
 
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
 
       - name: Test via ${{ matrix.goVer }}
         env:
-          STREAM_CHAT_API_KEY: ${{ secrets.STREAM_CHAT_API_KEY }}
-          STREAM_CHAT_API_SECRET: ${{ secrets.STREAM_CHAT_API_SECRET }}
+          STREAM_KEY: ${{ secrets.STREAM_CHAT_API_KEY }}
```
Comment (ferhatelmas) [inline .github/workflows/ci.yml:30]: himm are we adding this v5 changelog?

File: .golangci.yml [hunk 3]
```diff
@@ -36,37 +36,38 @@ run:
...
+  disable:
+   - wrapcheck
+   - varnamelen
+   - thelper
+   - tagliatelle
+   - paralleltest
+   - nlreturn
+   - exhaustivestruct
+   - wsl
+   - goerr113
```
Comment (ferhatelmas) [inline .golangci.yml:49]: handling this would be nice overall

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -20,7 +20,11 @@ We enforce code formatting with [`gufumpt`](https://github.com/mvdan/gofumpt) (a
 "editor.formatOnSave": true,
 "gopls": {
     "formatting.gofumpt": true
-}
+},
+"go.lintTool": "golangci-lint",
+"go.lintFlags": [
+    "--fast"
```
Comment (ferhatelmas) [inline CONTRIBUTING.md:26]: if this is given, does it really run all given linters? try with `-v` option to see

File: run-lint.sh [hunk 5]
```diff
@@ -1,16 +0,0 @@
...
-gopath="$(go env GOPATH)"
-
-if ! [[ -x "$gopath/bin/golangci-lint" ]]; then
-	echo >&2 'Installing golangci-lint'
-	curl --silent --fail --location \
-		https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$gopath/bin" v1.43.0
-fi
-
-# configured by .golangci.yml
-"$gopath/bin/golangci-lint" run
```
Comment (ferhatelmas) [inline run-lint.sh:16]: how are we going to run locally?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (peterdeme) [reply .github/workflows/ci.yml:30]: well, this is a quite new feature and I assume no one is using it yet: https://github.com/GetStream/stream-chat-go/commit/6cac452969917b7cbd2c8cfe34ca149b90046377#diff-4b667feae66c9d46b21b9ecc19e8958cf4472d162ce0a47ac3e8386af8bbd8cf

I think we can make an exception here and not handle this as a breaking change

File: CONTRIBUTING.md (hunk 4 above)
Comment (peterdeme) [reply CONTRIBUTING.md:26]: yes.
https://golangci-lint.run/usage/integrations/#editor-integration

File: http_test.go [hunk 6]
```diff
@@ -25,7 +25,7 @@ func TestRateLimit(t *testing.T) {
 			Messages: HardDelete,
 		})
 		if err != nil {
-			apiErr, ok := err.(Error)
+			apiErr, ok := err.(Error) //nolint:errorlint
```
[the remark this answers is not in the document]
Comment (peterdeme) [reply http_test.go:28]: never mind, I actually rather fixed this issue :) ✅

File: run-lint.sh (hunk 5 above)
Comment (peterdeme) [reply run-lint.sh:16]: Does Go have dev dependencies in `go.mod`? I'm searching for it in Google and I don't think so. 🤔 

By the way, when I've set up `"go.lintTool": "golangci-lint",` in VS Code, it automatically installed the tool.

Want me to add this file back? This was primarily used in the CI and I don't know if it makes sense to have it anymore.

File: CONTRIBUTING.md (hunk 4 above)
Comment (peterdeme) [reply CONTRIBUTING.md:26]: they recommend this, otherwise it can freeze your editor

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ferhatelmas) [reply .github/workflows/ci.yml:30]: master is already breaking, just not forget adding into changelog

File: .golangci.yml (hunk 3 above)
Comment (ferhatelmas) [reply .golangci.yml:49]: yes, keep for the next iteration https://github.com/GetStream/stream-chat-go/issues/9

File: run-lint.sh (hunk 5 above)
Comment (ferhatelmas) [reply run-lint.sh:16]: without this, hard to test locally

make is better of course, you can copy some parts from server

Comment (peterdeme) [reply run-lint.sh:16]: added a `Makefile`

File: CONTRIBUTING.md (hunk 4 above)
Comment (peterdeme) [reply CONTRIBUTING.md:26]: 👀 really? I can't find the typo

Comment (ferhatelmas) [reply CONTRIBUTING.md:26]: gufumpt

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+GOPATH := $(shell go env GOPATH)
+
+install-golangci:
+	test -s $(GOPATH)/bin/golangci-lint || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin
```
Comment (ferhatelmas) [inline Makefile:4]: * local install
* install automatically if need via make
* set a version

File: CONTRIBUTING.md (hunk 4 above)
Comment (peterdeme) [reply CONTRIBUTING.md:26]: ✅

File: Makefile (hunk 7 above)
Comment (peterdeme) [reply Makefile]: ✅

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**